crates/circuit/src/parameter/symbol_parser.rs

```diff
@@ -74,9 +76,9 @@ fn parse_symbol(s: &str) -> IResult<&str, SymbolExpr, VerboseError<&str>> {
                     // if array indexing is required in the future
                     // add indexing in Symbol struct
                     let s = format!("{v}[{i}]");
-                    Ok(SymbolExpr::Symbol(Arc::new(s)))
+                    Ok(SymbolExpr::Symbol(Symbol::new(&s, None, None)))
```

Comment:
Here we parse index of vector element, so Symbol can be initialized as
```suggestion
                    Ok(SymbolExpr::Symbol(Symbol::new(v, None, Some(i))))
```

Reply:
Good catch! Done 👍🏻